internal/pkg/keeper/client.go

```diff
@@ -168,7 +168,7 @@ func (client *keeperClient) WatchForChanges(updateChannel chan<- interface{}, er
 	var msgBusConfig models.MessageBusInfo
 	configStruct, ok := config.(*models.ConfigurationStruct)
 	if !ok {
-		configErr := errors.New("message bus information not defined in the configuration")
+		configErr := errors.New("configuration data conversion failed")
 		close(messages)
 		errorChannel <- configErr
 		return
```

Comment:
you need the check if configStruct.MessageQueue is empty
if it is empty, return the original error you made 
`errors.New("message bus information (MessageQueue section) is not defined in the configuration")`

Reply:
Since the `msgBusConfig` will be an empty struct(with default value) if no MsgQueue section is defined from configuration, check if the primary fields exist instead.